core/src/main/java/hudson/cli/CLICommand.java

```diff
@@ -253,38 +256,60 @@ public int main(List<String> args, Locale locale, InputStream stdin, PrintStream
             if (!(this instanceof HelpCommand || this instanceof WhoAmICommand))
                 Jenkins.get().checkPermission(Jenkins.READ);
             p.parseArgument(args.toArray(new String[0]));
-            LOGGER.log(Level.FINE, "Invoking CLI command {0}, with {1} arguments, as user {2}.",
-                    new Object[] {getName(), args.size(), auth.getName()});
+
+            Listeners.notify(
```

Comment:
`jenkins.util.Listeners` looks like a good option here. `asSystem=false`, the proposed interface pass along user authentication

Reply:
> `asSystem=false`, the proposed interface pass along user authentication

Could you explain this in detail? Seems wrong to not impersonate SYSTEM in the listener if you pass along the authentication anyway.

Reply:
perhaps I misinterpreted the javadoc on Listeners#asSystem.  Changing to asSystem=true